src/node.cc

```diff
@@ -3687,6 +3683,15 @@ void Init(int* argc,
     }
   }
 
+#ifdef __POSIX__
+  // Block SIGPROF signals when sleeping in epoll_wait/kevent/etc.  Avoids the
+  // performance penalty of frequent EINTR wakeups when the profiler is running.
+  // Only for v8.log profiling. CpuProfiler API expects to have evenly distributed samples.
```

Comment:
I'm fairly certain both the SDK and mingw know about SIGPROF but you're right, it's probably best to be conservative.